metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractExecutablePlugin.java

```diff
@@ -250,21 +228,46 @@ public MonitorResult monitor(DpsClient dpsClient) throws ExternalTaskException {
   void updateExecutionProgress(TaskInfo taskInfo) {
 
     // Calculate the various counts.
-    int expectedRecordCount = taskInfo.getExpectedRecordsNumber();
-    int processedRecordCount = taskInfo.getProcessedRecordsCount() +
-        taskInfo.getIgnoredRecordsCount() + taskInfo.getDeletedRecordsCount();
-    int deletedRecordCount = taskInfo.getDeletedRecordsCount();
+    // The expectedRecordsNumber we get from ecloud is dynamic and can change during execution.
+    int expectedRecordCount;
+    int processedRecordCount;
+    int deletedRecordCount;
+
+    if (getPluginMetadata() instanceof AbstractHarvestPluginMetadata && ((AbstractHarvestPluginMetadata) this.getPluginMetadata())
+        .isIncrementalHarvest()) {
+      //Incremental Harvest
+      //deletedRecordsCount never used
+      //expectedPostProcessedRecordsNumber and postProcessedRecordsCount represent deleted records
+      expectedRecordCount = taskInfo.getExpectedRecordsNumber();
+      processedRecordCount = taskInfo.getProcessedRecordsCount() + taskInfo.getIgnoredRecordsCount();
+      deletedRecordCount = taskInfo.getPostProcessedRecordsCount();
+    } else if (getPluginMetadata() instanceof AbstractHarvestPluginMetadata) {
+      //Full Harvest
+      //expectedPostProcessedRecordsNumber, postProcessedRecordsCount and ignoredRecordsCount not used
+      //deletedRecordsCount is always 0
+      expectedRecordCount = taskInfo.getExpectedRecordsNumber();
+      processedRecordCount = taskInfo.getProcessedRecordsCount();
+      deletedRecordCount = taskInfo.getDeletedRecordsCount();
+    } else if (getPluginMetadata() instanceof AbstractIndexPluginMetadata && !((AbstractIndexPluginMetadata) this
+        .getPluginMetadata()).isIncrementalIndexing()) {
+      //Full Indexing
+      //ignoredRecordsCount never used
+      //expectedPostProcessedRecordsNumber and postProcessedRecordsCount represent deleted records
+      //The deletedRecordsCount is always 0
+      expectedRecordCount = taskInfo.getExpectedRecordsNumber();
+      processedRecordCount = taskInfo.getProcessedRecordsCount();
+      deletedRecordCount = taskInfo.getPostProcessedRecordsCount();
+    } else {
+      //Other plugins including incremental indexing
+      //expectedPostProcessedRecordsNumber, postProcessedRecordsCount and ignoredRecordsCount not used
+      expectedRecordCount = taskInfo.getExpectedRecordsNumber() - taskInfo.getDeletedRecordsCount();
```

Comment:
I think the problem here is that the expected records count decreases gradually as the task progresses and it finds more and more deleted records. Are we sure we want this? To do this properly, we would need the number of expected deleted records.

Reply:
That is indeed true for all executions now(It used to be just for oai).
We discussed this and the solution would be another counter, but for now we decided to keep it this way(since it was somewhat possible for oai already) and see what the user experience would be.
Unless we decide otherwise.